### PR TITLE
Legends activation and deactivation on Line Chart

### PR DESCRIPTION
**Pull request by anant00729**
To test, 

1. go to http://localhost:3000/sensor/<location_id>/readings.
2. make sure to have sensor readings in the pg-litefarm local database.
3. check if you can see legends with a dot.
4. click on the legend and see if the graph shows or hides the line.

 
Screen Shot 2022-07-20 at 11 35 40 AM

Also, try adding some location ids as follows:

File Location: packages/webapp/src/containers/SensorReadings/index.jsx

Screen Shot 2022-07-20 at 11 21 33 AM

You'll be able to see this after adding multiple location id:

Screen Shot 2022-07-20 at 11 40 08 AM

After clicking on legends:

Screen Shot 2022-07-20 at 11 44 39 AM

Repeat the above steps to test activate/inactivate lines on the legends click.

For more information, please check 
https://lite-farm.atlassian.net/browse/LF-2502

### CODE REVIEW

**Review by Cali0707 — CHANGES_REQUESTED** [review]
@anant00729 from the description of the requirements for this, there should be checkboxes to enable/disable each line, but I'm not seeing any checkboxes here

**Comment by anant00729** [comment]
Right now, I don't have any visual inputs on how the checkboxes should appear. I am using recharts components to draw colored circles. As per the discussion that I had with Kevin, he told me to do it in the best way possible and then he ll be giving his inputs in case we need some more visual improvements.

File: packages/webapp/src/components/SensorReadingsLineChart/index.jsx [hunk 1]
```diff
@@ -25,12 +26,62 @@ const PureSensorReadingsLineChart = ({
...
+  const [legendsList, setLegendsList] = useState({});
 
-  const selectLine = (event) => {
-    if (event.dataKey.includes(NO_DATA)) return;
-    let sl = selectedLine === event.dataKey ? null : event.dataKey.trim();
-    setSelectedLine(sl);
+  useEffect(() => {
+    if (yAxisDataKeys.length) {
+      setLegendsList(
+        yAxisDataKeys.reduce((acc, cv, idx) => {
```
Comment (Cali0707) [inline packages/webapp/src/components/SensorReadingsLineChart/index.jsx:34]: We can probably get rid of `idx` as it doesn't look like it is being used in the reduce loop

**Review by Cali0707 — APPROVED** [review]
This is working as described for me. I let a comment about one small change that would clean up the code a little bit, but otherwise this looks good.

File: packages/webapp/src/components/SensorReadingsLineChart/index.jsx (hunk 1 above)
Comment (anant00729) [reply packages/webapp/src/components/SensorReadingsLineChart/index.jsx:34]: I am using it on line 38.

**Comment by anant00729** [comment]
I am using idx on line 38

File: packages/webapp/src/components/SensorReadingsLineChart/index.jsx (hunk 1 above)
Comment (smattingly) [reply packages/webapp/src/components/SensorReadingsLineChart/index.jsx:34]: :+1:

Comment (Cali0707) [reply packages/webapp/src/components/SensorReadingsLineChart/index.jsx:34]: Ah my bad, I missed that

**Pull request merged**